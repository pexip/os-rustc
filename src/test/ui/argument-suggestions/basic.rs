// Some basic "obvious" cases for the heuristic error messages added for #65853
// One for each of the detected cases

enum E { X, Y }
enum F { X2, Y2 }
struct G {}
struct H {}
struct X {}
struct Y {}
struct Z {}


fn invalid(_i: u32) {}
fn extra() {}
fn missing(_i: u32) {}
fn swapped(_i: u32, _s: &str) {}
fn permuted(_x: X, _y: Y, _z: Z) {}

fn main() {
    invalid(1.0); //~ ERROR mismatched types
    extra(""); //~ ERROR this function takes
    missing(); //~ ERROR this function takes
    swapped("", 1); //~ ERROR mismatched types
    //~^ ERROR mismatched types
    permuted(Y {}, Z {}, X {}); //~ ERROR mismatched types
    //~^ ERROR mismatched types
    //~| ERROR mismatched types

    let closure = |x| x;
    closure(); //~ ERROR this function takes
}
