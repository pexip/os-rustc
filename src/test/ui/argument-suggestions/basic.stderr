error[E0308]: mismatched types
  --> $DIR/basic.rs:20:13
   |
LL |     invalid(1.0);
   |             ^^^ expected `u32`, found floating-point number

error[E0061]: this function takes 0 arguments but 1 argument was supplied
  --> $DIR/basic.rs:21:5
   |
LL |     extra("");
   |     ^^^^^ -- supplied 1 argument
   |     |
   |     expected 0 arguments
   |
note: function defined here
  --> $DIR/basic.rs:14:4
   |
LL | fn extra() {}
   |    ^^^^^

error[E0061]: this function takes 1 argument but 0 arguments were supplied
  --> $DIR/basic.rs:22:5
   |
LL |     missing();
   |     ^^^^^^^-- supplied 0 arguments
   |     |
   |     expected 1 argument
   |
note: function defined here
  --> $DIR/basic.rs:15:4
   |
LL | fn missing(_i: u32) {}
   |    ^^^^^^^ -------

error[E0308]: mismatched types
  --> $DIR/basic.rs:23:13
   |
LL |     swapped("", 1);
   |             ^^ expected `u32`, found `&str`

error[E0308]: mismatched types
  --> $DIR/basic.rs:23:17
   |
LL |     swapped("", 1);
   |                 ^ expected `&str`, found integer

error[E0308]: mismatched types
  --> $DIR/basic.rs:25:14
   |
LL |     permuted(Y {}, Z {}, X {});
   |              ^^^^ expected struct `X`, found struct `Y`

error[E0308]: mismatched types
  --> $DIR/basic.rs:25:20
   |
LL |     permuted(Y {}, Z {}, X {});
   |                    ^^^^ expected struct `Y`, found struct `Z`

error[E0308]: mismatched types
  --> $DIR/basic.rs:25:26
   |
LL |     permuted(Y {}, Z {}, X {});
   |                          ^^^^ expected struct `Z`, found struct `X`

error[E0057]: this function takes 1 argument but 0 arguments were supplied
  --> $DIR/basic.rs:30:5
   |
LL |     closure();
   |     ^^^^^^^-- supplied 0 arguments
   |     |
   |     expected 1 argument

error: aborting due to 9 previous errors

Some errors have detailed explanations: E0057, E0061, E0308.
For more information about an error, try `rustc --explain E0057`.
